src/pg_to_evalscript/javascript_processes/quantiles.js

```diff
@@ -0,0 +1,99 @@
+class QuantilesParameterMissing extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "QuantilesParameterMissing";
+  }
+}
+
+class QuantilesParameterConflict extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "QuantilesParameterConflict";
+  }
+}
+
+function quantiles(arguments) {
+  const { data, probabilities, q, ignore_nodata = true } = arguments;
+
+  if (data === undefined || data === null) {
+    throw new Error("Mandatory argument `data` is either null or not defined");
+  }
+
+  if (!Array.isArray(data)) {
+    throw new Error("Argument `data` is not an array.");
+  }
+
+  if (typeof ignore_nodata !== "boolean") {
+    throw new Error("Argument `ignore_nodata` is not a boolean.");
+  }
+
+  if (probabilities === undefined && q === undefined) {
+    throw new QuantilesParameterMissing("The process `quantiles` requires either the `probabilities` or `q` parameter to be set.");
+  }
+
+  if (probabilities !== undefined && q !== undefined) {
+    throw new QuantilesParameterConflict("The process `quantiles` only allows that either the `probabilities` or the `q` parameter is set.");
+  }
+
+  let probs = [];
+
+  if (!!q && probabilities === undefined) {
```

Comment:
Minor: `&& probabilities === undefined` is redundant, given we already check if both are defined. Same goes for `&& q === undefined`.

Reply:
True, thank. I also changed `!!q` to `q !== undefined` and `!!probabilities` to `probabilities !== undefined` because `!null` and `!!0` are both `false` and if 0 or `null` would be passed for any of those params, the behaviour would be wrong.